include/ofi.h

```diff
@@ -98,6 +98,23 @@ extern "C" {
 
 #define ofi_div_ceil(a, b) ((a + b - 1) / b)
 
+#define ofi_val64_gt(x, y) \
+    (((int64_t) ((uint64_t) (x) - (uint64_t) (y))) > 0)
+#define ofi_val64_ge(x, y) \
+    (((int64_t) ((uint64_t) (x) - (uint64_t) (y))) >= 0)
+#define ofi_val64_lt(x, y) ofi_val64_gt(y, x)
+
+#define ofi_val32_gt(x, y) \
+    (((int32_t) ((uint32_t) (x) - (uint32_t) (y))) > 0)
+#define ofi_val32_ge(x, y) \
+    (((int32_t) ((uint32_t) (x) - (uint32_t) (y))) >= 0)
+#define ofi_val32_lt(x, y) ofi_val32_gt(y, x)
+
+#define ofi_val32_inrange(start, length, value) \
+    ofi_val32_ge(value, start) && ofi_val32_lt(value, start + length)
+#define ofi_val64_inrange(start, length, value) \
+    ofi_val64_ge(value, start) && ofi_val64_lt(value, start + length)
```

Comment:
Since we have to have specific versions for 32-bit vs 64-bit, I would convert these from macros into static inline calls, so that we can add type checks.  Would that still work given the macros below?

Reply:
Right, that'd be good to do so we get the warning if the wrong type is used. I can make `ofi_recvwin_id_valid()` a static inline as well and then this will work.

Reply:
I changed the macros to inline functions. Thank you for your help with cleaning up this check.